Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -557,6 +558,30 @@ public class PBXProjGenerator {
         return variantGroup
     }
 
+    func getBundle(atPath path: Path) -> SourceFile {
+        // It is possible for several targets to depend on the same bundle.
+        // Do not attempt to create a bundle for each target.
+        if let cachedBundle = bundleFiles[path] {
+            return cachedBundle
+        }
+        let fileReference = PBXFileReference(reference: generateUUID(PBXFileReference.self, path.lastComponent),
+                                             sourceTree: .group,
+                                             explicitFileType: "wrapper.plug-in",
+                                             path: path.lastComponent, includeInIndex: 1)
+        fileReferencesByPath[path] = fileReference.reference
+        addObject(fileReference)
+        // Add the bundle to the parent's group.
+        let group = getIntermediateGroups(
+            path: path.parent().parent(),
+            group: getSingleGroup(path: path.parent(), mergingChildren: [fileReference.reference], depth: 0)
+        )
+        addObject(group)
```

Comment:
`getIntermediateGroups` will recursively do `addObject(group)` for you (so you don't have to do it here)

Reply:
- yes
- this group isn't used as a child anywhere
- why is a group being created?
- intermedia groups should only be made if the spec.options.createIntermediateGroups == true

Reply:
This code uses `getIntermediateGroup` as a helper to get all the groups - it is unrelated the user facing feature. We need make sure all groups are created - otherwise it isn't created if there is no other files in the directory than the bundle. Does that make sense?

@bkase you are right, we shouldn't add parent group here.


Re